### PR TITLE
Manage students by index number

### PR DESCRIPTION
**Pull request by Deloryn**

### CODE REVIEW

File: Capri.Web/Controllers/StudentController.cs [hunk 1]
```diff
@@ -103,7 +114,7 @@ public IActionResult GetAll()
         }
 
         [AllowedRoles(RoleType.Dean)]
-        [HttpDelete("{id}")]
+        [HttpDelete("{id:Guid}")]
```
Comment (Arebisc) [inline Capri.Web/Controllers/StudentController.cs]: :Guid is unnecessery

File: Capri.Web/Controllers/StudentController.cs [hunk 2]
```diff
@@ -113,5 +124,17 @@ public async Task<IActionResult> Delete(Guid id)
             }
             return BadRequest(result.GetAggregatedErrors());
         }
+
+        [AllowedRoles(RoleType.Dean)]
+        [HttpDelete("{indexNumber:int}")]
```
Comment (Arebisc) [inline Capri.Web/Controllers/StudentController.cs]: :int unnecessery

File: Capri.Services/Students/StudentCreator.cs [hunk 3]
```diff
@@ -65,5 +73,15 @@ await _userCreator
             var studentViewModel = _mapper.Map<StudentViewModel>(student);
             return ServiceResult<StudentViewModel>.Success(studentViewModel);
         }
+
+        private async Task<bool> IsIndexNumberTaken(int indexNumber)
+        {
+            var student = await _context.Students.FirstOrDefaultAsync(s => s.IndexNumber == indexNumber);
```
[the remark this answers is not in the document]
Comment (Deloryn) [reply Capri.Services/Students/StudentCreator.cs]: Done

File: Capri.Services/Students/StudentDeleter.cs [hunk 4]
```diff
@@ -41,6 +42,27 @@ await _context
...
+                .Students
+                .FirstOrDefaultAsync(p => p.IndexNumber == indexNumber);
+
+            if(student == null)
+            {
+                return ServiceResult<StudentViewModel>.Error(
+                    $"Student with index number {indexNumber} does not exist");
+            }
+
+            return await Delete(student);
```
[the remark this answers is not in the document]
Comment (Deloryn) [reply Capri.Services/Students/StudentDeleter.cs]: Done

File: Capri.Services/Students/StudentUpdater.cs [hunk 5]
```diff
@@ -79,5 +87,19 @@ await _context
             var studentViewModel = _mapper.Map<StudentViewModel>(existingStudent);
             return ServiceResult<StudentViewModel>.Success(studentViewModel);
         }
+
+        private async Task<bool> IsIndexNumberTaken(Guid myId, int newIndexNumber)
+        {
+            var student = await _context.Students.FirstOrDefaultAsync(s => s.IndexNumber == newIndexNumber);
+            if(student == null)
```
[the remark this answers is not in the document]
Comment (Deloryn) [reply Capri.Services/Students/StudentUpdater.cs]: Done

File: Capri.Web.ViewModels/Student/StudentRegistration.cs [hunk 6]
```diff
@@ -15,6 +15,16 @@ public class StudentRegistration
         [MinLength(8)]
         [MaxLength(24)]
         public string Password { get; set; }
+
+        [Required]
+        [Range(100000, 999999)]
```
[the remark this answers is not in the document]
Comment (Deloryn) [reply Capri.Web.ViewModels/Student/StudentRegistration.cs]: Done

File: Capri.Web/Controllers/StudentController.cs [hunk 7]
```diff
@@ -28,7 +28,7 @@ public class StudentController : Controller
             _studentDeleter = studentDeleter;
         }
         
-        [HttpGet("{id}")]
+        [HttpGet("{id:Guid}")]
```
[the remark this answers is not in the document]
Comment (Deloryn) [reply Capri.Web/Controllers/StudentController.cs]: This is necessary in this case

File: Capri.Web/Controllers/StudentController.cs [hunk 8]
```diff
@@ -39,6 +39,17 @@ public async Task<IActionResult> Get(Guid id)
             return BadRequest(result.GetAggregatedErrors());
         }
 
+        [HttpGet("{indexNumber:int}")]
```
[the remark this answers is not in the document]
Comment (Deloryn) [reply Capri.Web/Controllers/StudentController.cs]: This is necessary in this case

File: Capri.Web/Controllers/StudentController.cs (hunk 1 above)
Comment (Deloryn) [reply Capri.Web/Controllers/StudentController.cs]: This is necessary in this case

File: Capri.Web/Controllers/StudentController.cs (hunk 2 above)
Comment (Deloryn) [reply Capri.Web/Controllers/StudentController.cs]: This is necessary in this case

File: Capri.Services/Students/StudentDeleter.cs [hunk 9]
```diff
@@ -42,7 +42,8 @@ await _context
                     $"Student with id {id} does not exist");
             }
 
-            return await Delete(student);
+            var deleteResult = await Delete(student);
+            return deleteResult;
```
Comment (Arebisc) [inline Capri.Services/Students/StudentDeleter.cs]: Here you can just `return Delete()`
Same situation I've told you about `return await` ;)

File: Capri.Services/Students/StudentDeleter.cs [hunk 10]
```diff
@@ -58,7 +59,8 @@ await _context
                     $"Student with index number {indexNumber} does not exist");
             }
 
-            return await Delete(student);
+            var deleteResult = await Delete(student);
+            return deleteResult;
```
Comment (Arebisc) [inline Capri.Services/Students/StudentDeleter.cs]: Same as above

File: Capri.Web/Controllers/StudentController.cs (hunk 7 above)
[the remark this answers is not in the document]
Comment (Arebisc) [reply Capri.Web/Controllers/StudentController.cs]: Why? Explain me

**Pull request merged**